source/_posts/2016-11-08-Vapor-setup.md

```diff
@@ -0,0 +1,311 @@
+
+Vapor will now ask you if you want to open the project. (Just go 'y')
+
+If you get 
+
+```jconsole
+ld: library not found for -lCLibreSSL for architecture x86_64
+clang: error: linker command failed with exit code 1 (use -v to see invocation)
+```
+
+Then just open terminal and run 
+```vapor build --clean```, after that run a ```vapor xcode``` that should do the trick.
+
+And hopefully you should be ready to rock and roll.
+
+### 2. Creating a model
+
+Open the terminal again, and run 
+```touch Source/App/Models/Car.swift```
+
+After that run ```vapor xcode``` this will make sure that you get the correct path for the model. You should now be able to see your model in the Models folder.
+
+Fill the model with the code below:
+
+```swift
+final class Car {
+    
+    //Add your properties like you normally would in iOS
+    var name:String
+    var color:String
+    var milesDriven:Int
+    
+    //Convinience method for instanciation of our object
+    init(name:String, color:String, milesDriven: Int) {
+        self.name = name
+        self.color = color
+        self.milesDriven = milesDriven
+    }
+}
+```
+### 3. Returning some JSON baby
+
+Now open the ```main.swift``` file and add the following code right above the ```drop.run()```
+
+```swift
+//Declaring our first route, this will be a 'get' (could also be 'post' etc..)
+drop.get("get_a_car") { request in
+    
+    //Creating our object
+    let car = Car(name: "Toyota", color: "Red", milesDriven: 0)
+    
+    //Formating it into JSON
+    return try JSON(node: [
+        "name" : car.name,
+        "color" : car.color,
+        "miles_driven" : car.milesDriven
+        ])
+}
+
+```
+Now Run your Xcode project. You should now see this in your console in Xcode
+
+```jconsole
+No command supplied, defaulting to serve...
+No preparations.
+Server 'default' starting at 0.0.0.0:8080
+```
+Now open your browser and type in ```0.0.0.0:8080/get_a_car```, you should now have returned your first JSON feed! 
+
+Ofc Vapor is way smarter than this. So please go ahead an delete the code you added in step 3.
+
+### 4. Creating a Controller
+
+Now go back to your terminal and run ```touch Sources/App/Controllers/CarController.swift``` and again ```vapor xcode```, the Controllers folder should now contain your newly created CarController.swift
+
+Before we can make any good use of our new Controller then we have to update our ```Car.swift``` model to accutally get some of Vapors awesome magic.
+
+So jump back into the ```Car.swift``` file and update the fill to have the following:
+
+```swift
+//Importing the Vapor framework to make the model support dependencies
+import Vapor
+//Importing Fluent which is Vapors powering of databases and tables
+import Fluent
+
+//Subclassing our Car model from 'Model' (Vapors magic model object)
+final class Car: Model {
+    
+    //Vapor uses 'Node' as their Model id's. This is the datatype that they use to make lookup and look at the primary key in the DB's.
```

Comment:
s/id's/ids. Love the comment, helps you understand what Node is and does. Something similar for Droplet would be nice :D